dozer/cogs/moderation.py

```diff
@@ -292,88 +298,133 @@ async def on_message(self, message):
-        elif len(message.content) == 0:
-            for i in message.embeds:
-                e.add_field(name="Title", value=i.title)
-                e.add_field(name="Description", value=i.description)
-                e.add_field(name="Timestamp", value=i.timestamp)
-                for x in i.fields:
-                    e.add_field(name=x.name, value=x.value)
-                e.add_field(name="Footer", value=i.footer)
+        audit = await self.check_audit(message.guild)
+        embed = discord.Embed(title="Message Deleted",
+                              description=f"Message Deleted In: {str(message.channel.mention)}\nSent by: {str(message.author.mention)}",
+                              color=0xFF0000, timestamp=message.created_at)
+        embed.set_thumbnail(url=message.author.avatar_url)
+        if audit:
+            if str(audit.target) == str(message.author.name + "#" + message.author.discriminator):
+                auditMember = await message.guild.fetch_member(audit.user.id)
+                embed.add_field(name="Message Deleted By: ", value=str(auditMember.mention), inline=False)
+        if message.content:
+            embed.add_field(name="Message Content:", value=str(message.content[0:1023]), inline=False)
+            if len(message.content) > 1024:
+                embed.add_field(name="Message Content Continued:", value=message.content[1024:2000], inline=False)
+        else:
+            embed.add_field(name="Message Content:", value="N/A", inline=False)
+        embed.set_footer(text="UserID:" + str(message.author.id))
         if message.attachments:
-            e.add_field(name="Attachments", value=", ".join([i.url for i in message.attachments]))
-        messagelogchannel = await self.edit_delete_config.query_one(guild_id=message.guild.id)
-        if messagelogchannel is not None:
-            channel = message.guild.get_channel(messagelogchannel.messagelog_channel)
+            embed.add_field(name="Attachments", value=", ".join([i.url for i in message.attachments]))
+        message_log_channel = await self.edit_delete_config.query_one(guild_id=message.guild.id)
+        if message_log_channel is not None:
+            channel = message.guild.get_channel(message_log_channel.messagelog_channel)
             if channel is not None:
-                await channel.send(embed=e)
+                await channel.send(embed=embed)
+
+    @Cog.listener()
+    async def on_raw_message_edit(self, payload):
+        """Logs message edits that are not currently in the bots message cache"""
+        if payload.cached_message:
+            return
+        Mchannel = self.bot.get_channel(int(payload.channel_id))
+        guild = Mchannel.guild
+        print(payload)
+        try:
+            content = payload.data['content']
+        except KeyError:
+            content = None
+        try:
+            author = payload.data['author']
+        except KeyError:
+            return
```

Comment:
When does `payload.data['author']` not exist?

Reply:
Never...

Reply:
In that case it's probably best to let `payload.data['author']` raise its error so that we get an error report instead of failing silently if that ever changes.